### PR TITLE
feat(text-area): add experimental styling

### PR DESCRIPTION
**Pull request by jnm2377**
Doesn't close an issue, this component was missed, I guess? 
Referenced Text Input #1218 

#### Changelog
Added experimental Text Area

**New**

- Experimental Text Area component

**Changed**

Moved `@include exports('text-area)` from line 15 to line 183, `@mixin text-area` now on line 15.

### CODE REVIEW

File: src/components/text-area/_text-area.scss [hunk 1]
```diff
@@ -109,3 +109,81 @@
...
+  //-----------------------------
+  // Disabled & Error icon spacing
+  //-----------------------------
+  .#{$prefix}--text-area:disabled,
+  .#{$prefix}--text-area[data-invalid],
+  .#{$prefix}--text-area:invalid,
+  .#{$prefix}--text-area[data-loading] {
+    padding-right: rem(40px);
+    background-repeat: no-repeat;
+    background-position: right 8px bottom 50%;
```
Comment (tw15egan) [inline src/components/text-area/_text-area.scss]: More of a question for @IBM/carbon-designers (unless you've already asked), is the error icon supposed to be centered or aligned to the top right / bottom right? 

screen shot 2018-11-05 at 3 08 55 pm

**Review by tw15egan — APPROVED** [review]
Looks great! I just had one small comment.

 Congrats on your first PR 🎉 🚀

**Comment by alisonjoseph** [comment]
Thank you for adding this! 🙌 Could you copy over the skeleton state styles to the experimental component also?

File: src/components/text-area/_text-area.scss (hunk 1 above)
Comment (jnm2377) [reply src/components/text-area/_text-area.scss]: Totally good question! Wasn't sure about that, but can definitely change to whatever it needs to be.

**Comment by jnm2377** [comment]
@alisonjoseph you got it! Just added them.

**Review by alisonjoseph — APPROVED** [review]
LGTM once the design question is answered! 🎉

**Review by asudoh — APPROVED** [review]
Congrats on your first PR @jnm2377! LGTM 👍 once design review completes.

File: src/components/text-area/_text-area.scss (hunk 1 above)
Comment (jeanservaas) [reply src/components/text-area/_text-area.scss]: Top-aligned, vertical center with the text just like notifications is the rule for icons... (although we're doing an audit of these error states and error icons this week)

File: src/components/text-area/_text-area.scss [hunk 2]
```diff
@@ -109,3 +109,111 @@
...
+    @include skeleton;
+    height: rem(100px);
+
+    &::-webkit-input-placeholder {
+      color: transparent;
+    }
+  }
+
+  // Skeleton State
+  #{$prefix}--text-area.#{$prefix}--skeleton {
```
Comment (tw15egan) [inline src/components/text-area/_text-area.scss]: Seems like a duplicate?

File: src/components/text-area/_text-area.scss [hunk 3]
```diff
@@ -109,3 +109,111 @@
...
```
Comment (tw15egan) [inline src/components/text-area/_text-area.scss:162]: Same as above ^

File: src/components/text-area/_text-area.scss (hunk 2 above)
Comment (jnm2377) [reply src/components/text-area/_text-area.scss]: You're right. Fixed these issues. Idk how I missed that. Thanks!

**Comment by jeanservaas** [comment]
### Comments on Text Area:

- Let's remove all icons from Text Area (both error icon and the grayed out disabled icon)... 1) we don't have them in Text Area in Carbon and 2) we're still trying to figure out where we're going to place error icons, or if we even have them at all in v10

- Error message should function just like it does in Text input, so it should be below the field (use the 4px spacer)

Yours:
screen shot 2018-11-09 at 12 46 16 pm

Spec:
screen shot 2018-11-09 at 12 47 31 pm

- Also, we found an inconsistency in the way we treat option helper text, it was different for number input and text input... @tw15egan is fixing this now... below is the new style and spacing spec for optional helper text (Note: optional helper text is true italic from now on to differentiate itself from label

New Spec:

screen shot 2018-11-09 at 1 07 37 pm

- I just opened up an issue for input field changes, we're going to kill the hover state on all input fields; we're also changing the default height of the input field from 32px to 40px

When you move the field around and take it back to minimum height, the text is no longer vertically centered, it needs to be vertically centered in the default field height... and then stay there as the field grows and shrinks, see grab below:

screen shot 2018-11-09 at 1 13 15 pm

- Also hint text looks like it's calling a different font, not Plex? Could we just double check that?

**Comment by jnm2377** [comment]
@jeanservaas I made the changes on text-area from your feedback, here's the updated [staging link](http://carbon-dev-environment-textarea-x-friendly-gazelle.mybluemix.net/?nav=text-area) for you to review.  I also changed the min-height to `40px` since I saw that TJ updated all the other inputs to `40px` in his PR, so the placeholder text will be vertically centered at the default height.

**Comment by jeanservaas** [comment]
This looks good to me... one note... we're still working on our type styles, but in the component review yesterday, Mike wanted us to add a type class for **12pt / 14pt** line height for the Optional Helper Text situation. Right now, it's spec'd at 12pt / 16pt.

- **Lauren is creating a new issue for this so maybe we go ahead an merge... we can apply it later once we nail down the type styles. Also, Anna thinks ultimately we're going to ditch the line height ratios and go with hard pixel values too, because the rem ratios aren't really translating.** 

This is what I'm talking about (the line height on that helper text):

image

**Pull request merged**